python/amici/ode_export.py

```diff
@@ -2803,3 +2809,20 @@ def csc_matrix(matrix: sp.Matrix,
 
     return symbol_col_ptrs, symbol_row_vals, sparse_list, symbol_list, \
         sparse_matrix
+
+
+def is_valid_identifier(x: str) -> bool:
+    """Check whether `x` is a valid identifier
+
+    Check whether `x` is a valid identifier for conditions, parameters,
+    observables... . Identifiers may only contain upper and lower case letters,
+    digits and underscores, and must not start with a digit.
+
+    Arguments:
+        x: string to check
+
+    Returns:
+        ``True`` if valid, ``False`` otherwise
+    """
+
+    return re.match(r'^[a-zA-Z_]\w*$', x) is not None
```

Comment:
`re.match` only searches at beginning of string
```suggestion
    return re.search(r'^[a-zA-Z_]\w*$', x) is not None
```

Reply:
But we explicitly match beginning and end (^, $)

Reply:
ah, right :)